controllers/operatorconfigmap_controller.go

```diff
@@ -698,10 +720,35 @@ func (c *OperatorConfigMapReconciler) reconcileSubscription() error {
 	return nil
 }
 
+func (c *OperatorConfigMapReconciler) reconcileWebhookService() error {
+	svc := &corev1.Service{}
+	svc.Name = templates.WebhookServiceName
+	svc.Namespace = c.OperatorNamespace
+	err := c.createOrUpdate(svc, func() error {
+		utils.AddAnnotation(svc, "service.beta.openshift.io/serving-cert-secret-name", "webhook-cert-secret")
+		templates.WebhookService.Spec.DeepCopyInto(&svc.Spec)
+		// the service is already deployed as part of olm bundle as that'll generate certs which is required to start manager.
+		// in reconcile we are making sure that service doesn't get deleted as long as the configmap stays by owning the service.
+		return c.own(svc)
```

Comment:
To follow with out conventions, owning the resource should be the first thing to do inside the `CreateOrUpdate` mutator.

Reply:
ya, I mistook that I was doing a deep copy of entire resource, will change.